drivers/adc/adc_smartbond_sdadc.c

```diff
@@ -254,26 +305,83 @@ static int sdadc_smartbond_read_async(const struct device *dev,
 }
 #endif /* CONFIG_ADC_ASYNC */
 
-static int sdadc_smartbond_init(const struct device *dev)
+static int sdadc_smartbond_resume(const struct device *dev)
 {
-	int err;
-	struct sdadc_smartbond_data *data = dev->data;
+	int ret;
 	const struct sdadc_smartbond_cfg *config = dev->config;
 
-	SDADC->SDADC_CTRL_REG = SDADC_SDADC_CTRL_REG_SDADC_EN_Msk;
-	SDADC->SDADC_CLEAR_INT_REG = 0x0;
+	da1469x_pd_acquire(MCU_PD_DOMAIN_COM);
+
 	SDADC->SDADC_TEST_REG =
 		(SDADC->SDADC_TEST_REG & ~SDADC_SDADC_TEST_REG_SDADC_CLK_FREQ_Msk) |
 		(config->sdadc_clk_freq) << SDADC_SDADC_TEST_REG_SDADC_CLK_FREQ_Pos;
+
 	/*
 	 * Configure dt provided device signals when available.
 	 * pinctrl is optional so ENOENT is not setup failure.
 	 */
-	err = pinctrl_apply_state(config->pcfg, PINCTRL_STATE_DEFAULT);
-	if (err < 0 && err != -ENOENT) {
-		LOG_ERR("ADC pinctrl setup failed (%d)", err);
-		return err;
+	ret = pinctrl_apply_state(config->pcfg, PINCTRL_STATE_DEFAULT);
+	if (ret < 0 && ret != -ENOENT) {
+		LOG_ERR("ADC pinctrl setup failed (%d)", ret);
```

Comment:
`PD_COM` should be released in case of error, isn't it?

Reply:
Good catch, added